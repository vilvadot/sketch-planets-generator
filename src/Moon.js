  // ------------ Seed --------------
  //  1      23        4           5-10
  //  size   period    distance    ??? 

export class Moon {

  constructor(seed, parentSize){
    this.seed = seed.toString()
    this.parentSize = parentSize
    this.orbitalPeriod = this.generateOrbitalPeriod()
    this.size = this.generateSize()
    this.distance = this.generateDistance()
  }

  generateSize(){
    const seed = this.seed[0]
    return seed * 0.8;
  }

  generateOrbitalPeriod(){
    const lastTwoDigits = this.seed.slice(1,3)
    
    return parseInt(lastTwoDigits / 2)
  }

  generateDistance(){
    const seed = this.seed.slice(3,4)
    return (this.parentSize + this.size) * seed * 1.1;
  }
}