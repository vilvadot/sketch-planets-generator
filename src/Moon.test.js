import { Moon } from "./Moon";
import { Random } from './Random'

describe("Moon", () => {
  it("can be recreated from a seed", () => {
    const aSeed = new Random(1234678930).next();
    const aParentSize = 4

    const moon = new Moon(aSeed, aParentSize)

    expect(moon).toMatchSnapshot();
  });
});