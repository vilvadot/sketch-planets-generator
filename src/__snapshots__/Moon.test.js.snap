// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Moon can be recreated from a seed 1`] = `
Moon {
  "distance": 10.56,
  "orbitalPeriod": 7,
  "parentSize": 4,
  "seed": "114295549",
  "size": 0.8,
}
`;
